sympy/functions/special/bessel.py

```diff
@@ -698,64 +698,91 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
-                return o
-            t = (_mexpand(r**2) + o).removeO()
+                return Order(z**(-nu) + z**nu, x)
 
-            if nu > S.Zero:
-                term = r**(-nu)*factorial(nu - 1)/2
-                b.append(term)
-                for k in range(1, nu):
-                    denom = (k - nu)*k
-                    if denom == S.Zero:
-                        term *= t/k
-                    else:
-                        term *= t/denom
-                    term = (_mexpand(term) + o).removeO()
-                    b.append(term)
+            o = Order(x**n, x)
+            if nu.is_integer:
+                # Reference: https://functions.wolfram.com/Bessel-TypeFunctions/BesselK/06/01/04/01/02/0008/ (only for integer order)
+                newn = ceiling(n/exp)
+                bn = besseli(nu, z)
+                a = ((-1)**(nu - 1)*log(z/2)*bn)._eval_nseries(x, n, logx, cdir)
 
-            p = r**nu*(-1)**nu/(2*factorial(nu))
-            term = p*(digamma(nu + 1) - S.EulerGamma)
-            c.append(term)
-            for k in range(1, (newn + 1)//2):
-                p *= t/(k*(k + nu))
-                p = (_mexpand(p) + o).removeO()
-                term = p*(digamma(k + nu + 1) + digamma(k + 1))
+                b, c = [], []
+                t = (_mexpand(r**2) + o).removeO()
+
+                if nu > S.Zero:
+                    term = r**(-nu)*factorial(nu - 1)/2
+                    b.append(term)
+                    for k in range(1, nu):
+                        denom = (k - nu)*k
+                        if denom == S.Zero:
+                            term *= t/k
+                        else:
+                            term *= t/denom
+                        term = (_mexpand(term) + o).removeO()
+                        b.append(term)
+
+                p = r**nu*(-1)**nu/(2*factorial(nu))
+                term = p*(digamma(nu + 1) - S.EulerGamma)
                 c.append(term)
-            return a + Add(*b) + Add(*c) # Order term comes from a
+                for k in range(1, (newn + 1)//2):
+                    p *= t/(k*(k + nu))
+                    p = (_mexpand(p) + o).removeO()
+                    term = p*(digamma(k + nu + 1) + digamma(k + 1))
+                    c.append(term)
+                return a + Add(*b) + Add(*c) # Order term comes from a
+            elif nu.is_noninteger:
+                # Reference: https://functions.wolfram.com/Bessel-TypeFunctions/BesselK/06/01/04/01/01/0003/
+                # (only for non-integer order).
+                # While the expression in the reference above seems correct
+                # for non-real order as well [e.g. Dunster 1990. "Bessel functions
+                # of purely imaginary order, with an application to second-order
+                # linear differential equations having a large parameter".
+                # SIAM J. Math. Anal. Vol 21, No. 4, pp 995-1018.], the
+                # implementation here assumes nu is real.
```

Comment:
Perhaps rather than saying "the implementation here assumes nu is real" this could say "if nu is not real then the expression would not represent a power series because it would have powers of x with non-real exponent."

Reply:
Done. New wording is "While the expression in the reference above seems correct for non-real order as well, it would need some manipulation (not implemented) to be written as a power series in x with real exponents".